lib/src/enforcer.dart

```diff
@@ -8,5 +14,37 @@ class Enforcer extends CoreEnforcer {
   /// [policyFile] is the path of the policy file.
   /// [adapter] is the adapter.
   /// [enableLog] whether to enable Casbin's log.
-  Enforcer(var model, String policyFile, Adapter adapter, bool enableLog);
+  Enforcer(
+      {String modelPath,
+      Model model,
+      String policyFile,
+      Adapter adapter,
+      bool enableLog})
+      : super(
+          model: model,
+          modelPath: modelPath,
+        );
+
+  Enforcer.fromModelPathAndPolicyFile(String modelPath, String policyFile) {
+    this.modelPath = modelPath;
+    final fileAdapter = FileAdapter(policyFile);
+    Enforcer.fromModelPathAndAdapter(modelPath, fileAdapter);
+  }
+
+  Enforcer.fromModelPathAndAdapter(String modelPath, Adapter adapter) {
+    this.modelPath = modelPath;
+    final model = Model();
+    model.loadModel(modelPath);
+    Enforcer.fromModelAndAdapter(model, adapter);
+  }
```

Comment:
You could use factory constructors instead to redirect constructors.

Reply:
@KNawm I'm not able to use factory constructors as _Instance members can't be accessed from a factory constructor_.

Reply:
That's correct, That's why I asked if the `Enforcer` constructor was public API or not, one way to achieve this is to use that constructor or you could create a new private constructor `Enforcer._()` to delegate the work of creating the class instance. This way you would provide all of the parameters to the main constructor and you won't interact with instance members on those methods.

Reply:
Alright! I'll do this.